app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -236,6 +238,7 @@ public void onCreate() {
             Logger.registerLogger(new AndroidLogger(this.getGlobalStorage(AndroidLogEntry.STORAGE_KEY, AndroidLogEntry.class)));
             pil.dumpToNewLogger();
         }
+        XPathErrorLogger.registerStorage(getGlobalStorage(XPathErrorEntry.STORAGE_KEY, XPathErrorEntry.class));

```

Comment:
I updated this PR to put the new XPath error logging in the user db. I will leave the work of moving normal logs from the global to user/app db for another PR.